src/com/com_nrf24l01_spi_protocol.cpp

```diff
@@ -0,0 +1,106 @@
+  auto spi_transfer_length = 2;
+  std::vector<std::uint8_t> mosi_data_buffer;
+  std::vector<std::uint8_t> miso_data_buffer(spi_transfer_length);
+  mosi_data_buffer.push_back(instruction_word::R_REGISTER | register_address);
+
+  spi_.Transfer(mosi_data_buffer, miso_data_buffer);
+
+  return miso_data_buffer.at(position_in_buffer);
+}
+
+auto NRF24L01SpiProtocol::ReadRegister(const std::uint8_t register_address, std::uint8_t length) noexcept -> std::vector<std::uint8_t> {
+  std::vector<std::uint8_t> mosi_data_buffer;
+  std::vector<std::uint8_t> miso_data_buffer(length + 1);
+  mosi_data_buffer.push_back(instruction_word::R_REGISTER | register_address);
+
+  spi_.Transfer(mosi_data_buffer, miso_data_buffer);
+  miso_data_buffer.erase(miso_data_buffer.begin());
+
+  return miso_data_buffer;
+}
+
+auto NRF24L01SpiProtocol::WriteRegister(std::uint8_t register_address, std::uint8_t register_content) noexcept -> types::DriverStatus {
+  std::vector<std::uint8_t> mosi_data_buffer;
+  mosi_data_buffer.push_back(instruction_word::W_REGISTER | register_address);
+  mosi_data_buffer.push_back(register_content);
+
+  auto spi_ret_val = spi_.Write(mosi_data_buffer);
+
+  return spi_ret_val;
+}
+
+auto NRF24L01SpiProtocol::WriteRegister(std::uint8_t register_address, std::vector<std::uint8_t> &register_content) noexcept -> types::DriverStatus {
+  std::vector<std::uint8_t> mosi_data_buffer;
+  mosi_data_buffer = register_content;
+  mosi_data_buffer.insert(mosi_data_buffer.begin(), (instruction_word::W_REGISTER | register_address));
+
+  auto spi_ret_val = spi_.Write(mosi_data_buffer);
+
+  return spi_ret_val;
+}
+
+auto NRF24L01SpiProtocol::WritePayloadData(std::vector<std::uint8_t> &payload) noexcept -> types::DriverStatus {
+  std::vector<std::uint8_t> mosi_data_buffer;
+  mosi_data_buffer = payload;
+  mosi_data_buffer.insert(mosi_data_buffer.begin(), instruction_word::W_TX_PAYLOAD);
+
+  auto spi_ret_val = spi_.Write(mosi_data_buffer);
+
+  return spi_ret_val;
+}
+
+auto NRF24L01SpiProtocol::ReadPayloadData(std::vector<std::uint8_t> &payload) noexcept -> types::DriverStatus {
+  std::vector<std::uint8_t> mosi_data_buffer;
+  mosi_data_buffer.push_back(instruction_word::R_RX_PAYLOAD);
+
+  auto spi_ret_val = spi_.Transfer(mosi_data_buffer, payload);
+  payload.erase(payload.begin());
+
+  return spi_ret_val;
+}
+
+auto NRF24L01SpiProtocol::FlushTxBuffer() noexcept -> types::DriverStatus {
+  std::vector<std::uint8_t> mosi_data_buffer;
+  mosi_data_buffer.push_back(instruction_word::FLUSH_TX);
+
+  auto spi_ret_val = spi_.Write(mosi_data_buffer);
+
+  return spi_ret_val;
+}
+
+auto NRF24L01SpiProtocol::FlushRxBuffer() noexcept -> types::DriverStatus {
+  std::vector<std::uint8_t> mosi_data_buffer;
+  mosi_data_buffer.push_back(instruction_word::FLUSH_RX);
+
+  auto spi_ret_val = spi_.Write(mosi_data_buffer);
+
+  return spi_ret_val;
+}
+
+auto NRF24L01SpiProtocol::ReadAndClearIRQFlags() noexcept -> register_t {
+  auto position_in_buffer = 0;
```

Comment:
please have a look at this variable and other variables in your functions that can easily by constexpr auto

Reply:
Done.